### PR TITLE
Implement API `api/envs/default/apps/

### PR DESCRIPTION
**Pull request by zzxwill**
implement API `api/envs/default/apps/ and refactor code

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -1,60 +1,7 @@
 package main
```
Comment (wonderflow) [inline cmd/server/main.go:1]: if we have `vela dashboard` command, we don't need this package any more

File: pkg/cmd/dashboard.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+by Cli and called by cmd/server/main.go.
+*/
+func SetupApiServer(kubeClient client.Client) {
+	var logFilePath string
+	var logRetainDate int
+	var logCompress, development bool
+
+	flag.StringVar(&logFilePath, "log-file-path", "", "The log file path.")
+	flag.IntVar(&logRetainDate, "log-retain-date", 7, "The number of days of logs history to retain.")
+	flag.BoolVar(&logCompress, "log-compress", true, "Enable compression on the rotated logs.")
```
Comment (wonderflow) [inline pkg/cmd/dashboard.go:60]: we don't need these arguments for command line, just use stdout

File: pkg/server/handler/appHandlers.go [hunk 3]
```diff
@@ -26,6 +32,14 @@ func GetApps(c *gin.Context) {
 }
 
 func ListApps(c *gin.Context) {
+	kubeClient := c.MustGet("KubeClient")
+	namespace := c.Param("envName")
```
Comment (wonderflow) [inline pkg/server/handler/appHandlers.go:36]: it's env name,not namespace. Namespace is one information contained in env which can retrieve from envfile by envName.

File: cmd/server/main.go (hunk 1 above)
Comment (zzxwill) [reply cmd/server/main.go:1]: Indeed. I still keep it as it can be the access to quickly launch API server. Let me delete it. Thanks.

Comment (wonderflow) [reply cmd/server/main.go:1]: we can add a Makefile command for quick launch

File: pkg/cmd/dashboard.go (hunk 2 above)
Comment (ryanzhang-oss) [reply pkg/cmd/dashboard.go:60]: this is for the RESTful server logging, won't this be good to have? We don't want to print those logs to the stdout, right?

Comment (wonderflow) [reply pkg/cmd/dashboard.go:60]: If this server is deployed in K8s cluster, print to stdout is enough.
If this is used as cli dashboard command, print to sdtout is also fine.

File: pkg/server/handler/appHandlers.go [hunk 4]
```diff
@@ -26,6 +32,14 @@ func GetApps(c *gin.Context) {
...
+	ctx := context.Background()
```
[the remark this answers is not in the document]
Comment (zzxwill) [reply pkg/server/handler/appHandlers.go]: Fixed.

File: pkg/server/handler/appHandlers.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zzxwill) [reply pkg/server/handler/appHandlers.go]: Done.

Comment (zzxwill) [reply pkg/server/handler/appHandlers.go]: Fixed, thanks.

**Comment by zzxwill** [comment]
Please ignore failure of build step `Install OAM runtime and prepare WorkloadDefinitions/TraitDefinitions`, I will fix it after https://github.com/cloud-native-application/RudrX/pull/132 is merged. Thanks.

File: pkg/cmd/dashboard.go (hunk 2 above)
Comment (ryanzhang-oss) [reply pkg/cmd/dashboard.go]: I don't get it, users don't want to see our server log printing out to their Cli command line, right?

File: pkg/server/handler/appHandlers.go [hunk 5]
```diff
@@ -26,6 +33,21 @@ func GetApps(c *gin.Context) {
 }
 
 func ListApps(c *gin.Context) {
+	kubeClient := c.MustGet("KubeClient")
+	envName := c.Param("envName")
+	envMeta, err := oam.GetEnvByName(envName)
```
Comment (ryanzhang-oss) [inline pkg/server/handler/appHandlers.go:38]: so looks like the envMeta is stored on the local file system?

Comment (zzxwill) [reply pkg/server/handler/appHandlers.go:38]: Fixed.

**Comment by zzxwill** [comment]
@ryanzhang-oss I like your integration test framework very much. But it will take me some time to have it passed as some fundamental apis like env create, system init needed to be implemented first. 
And will you consider move it to https://github.com/cloud-native-application/RudrX/tree/master/e2e and named a folder `apiserver` to cover all api server test cases?

**Pull request merged**